### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by knightburton**
IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

**Comment by knightburton** [comment]
@daeyeon Altough this is a WIP PR, I would like to know that what do you think about this and also It would be great if you can help me to improve this.

**Comment by daeyeon** [comment]
@knightburton Thanks for asking my thought. In my first glance, I'm a bit concerned that since the document is in detail, it wouldn't always include up-to-date information during the development. On the other hand, the details would give user good impressions and helpful. One thing I'd like to suggest is to remove the screenshot image from the `Layout overview` section. Instead, we can show the latest running demo through  https://samsung.github.io/iotjscode/. 

Regarding FOSSA, I guess you want me to set up FOSSA and to add its badge on this. I can surely take that.

**Comment by knightburton** [comment]
@daeyeon Thank you for your thought. I know this is a bit detailed, at least the information panels descriptions, I thought that we should give the users some information about those panels a bit. Also, If you think that less information would be better, like what is the purpose of each panel without button description, I am fine with that because you are right, we are in dev stage.

Yes, if you can take care the FOSSA part that would be great for me.

**Comment by daeyeon** [comment]
@knightburton We don't need to rewrite what is already well-written. Let's land the current version, and fix the strategy according to the future situation if needed.

Setting up FOSSA for iotjscode is completed with no issue. Please use the below md snippet to put the badge.

```
[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2FSamsung%2Fiotjscode.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2FSamsung%2Fiotjscode?ref=badge_shield)
```

**Comment by knightburton** [comment]
@daeyeon Thank you and I will remove the [WIP] tag then.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,170 @@
...
+
+# Layout overview
+Latest [Demo Page](https://samsung.github.io/iotjscode/).
+
+There are four section in the `IoT.js Code`:
+1. **Menu**
+
+    The menu contains three more section:
+    - The hamburger style **menu toggle button**, which is for to close or open the whole menu. In closed state the menu item texts are not visible, just the icons.
+    - The **File**, **Settings** and **Download** items. In the File menu, you can open, create or save a file. In the Settings menu you can modify the editor settings or you can reset the settings to the default values. Finally in Download manu you can download the Memory Usage report if that is available.
```
Comment (daeyeon) [inline README.md]: manu -> menu

For good readability, I suggest putting commas like: 
a) In the Settings menu, you can ...
b) Finally, in Download menu, you can download ....
c) settings, or you can...

File: README.md [hunk 2]
```diff
@@ -1,2 +1,170 @@
...
+    - The **panel toggle buttons**, these are for to turn on or off a single panel on the information panels area.
+
+2. **Action buttons**
+
+    These are on the top of the page. With these buttons and inputs you can connect to the engine and you can control the main debugging process. The default `ip` and `port` where the client will try to connect is `localhost:5001`.
```
Comment (daeyeon) [inline README.md]: I suggest comma here : With these buttons and inputs, ..

File: README.md [hunk 3]
```diff
@@ -1,2 +1,170 @@
...
+
+3. **Information panels**
+
+    These panels are on the left side of the workspace. Whitin these panels you can see a few really useful information about the current state of the debugging.
```
Comment (daeyeon) [inline README.md]: Whitin -> Within

File: README.md [hunk 4]
```diff
@@ -1,2 +1,170 @@
...
+    The available panels are the following:
+    - **Backtrace panel**: when the debugger is stopped (on a breakpoint or an exception) this will contain the call stack from the stopped statement.
```
Comment (daeyeon) [inline README.md]: statement -> state?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,170 @@
...
+    - **Watch panel**: you can add a variable or any expression what you want to watch while your code running. You can add a new expression with the `Add` button in the panel header, you can `Refresh` the watch list or you can delete every watched expression with the `Clear` button. If you do not want to delete all expression at once then you can delete them one-by-one with the `minus` icon at the end of an expression line in the list.
+    - **Memory Chart panel**: you can monitoring the memory usage of your code under running. Each time when the debugger stopped, the the memory chart will be updated with new values. The chart will show you the **total size** of your memory usage as well as the **byte code** size, **string** size, **property** size, **object** size and **allocated** memory size. You can control the panel behaviour with the buttons in the panel header. The `Clear` button will reset the whole diagram, the `Pause` button will suspend the data collection and the `Record` button will start (or restart) the process.
```
Comment (daeyeon) [inline README.md]: - the the memory -> the
- behaviour -> behavior

File: README.md [hunk 6]
```diff
@@ -1,2 +1,170 @@
...
+    - **Output log panel**: in this panel you can see what is your code output. Every text based output will be displayed in this.
```
Comment (daeyeon) [inline README.md]: suggest adding hyphen: text-based

File: README.md [hunk 7]
```diff
@@ -1,2 +1,170 @@
...
+    - **Source sending panel**: this section allows you to send source code to the engine on-the-fly after the engine is started in waiting mode. The available sources (the currently opened files) will be listed in the panel left side and you can select those file what you want to upload and run on the device. The selected files will be listed on the panel right side and the selected files order can be modified by grab and move them up or down. After you selected and ordered the files you can use the control buttons in the panel header. The `Run` button will send the sources to the device (if the engine is in waiting mode), the `Context reset` button will reset the engine context and will broke the connection between the engine and the client and will try to reconnect. The `Clear` button will remove the selected files from the stack. **Note** that the source sender process is sending one file at a time and the context reset will be processed after a file sending is done and the engine is waiting for a new source file.
```
Comment (daeyeon) [inline README.md]: broke -> break

File: README.md [hunk 8]
```diff
@@ -1,2 +1,170 @@
...
+    - **Console panel**: you can follow the whole debugging work in this panel. Every action or information will be displayed in this as plain text. There are a few basic command which are available in the console, if you want to use them just type the `help` keyword into the command line input. This panel has a unique feature, when you connected to a device and the code which is running on that is not loaded into the editor, it will give you a button, `Load from Jerry`, which is able to load the running source code from the device into a new file in the editor.
```
Comment (daeyeon) [inline README.md]: are -> is (which is available)

File: README.md [hunk 9]
```diff
@@ -1,2 +1,170 @@
...
+
+
+4. **Editor**
+
+    The editor can be devided into two section. The first one is the `file header` where you can switch between the opened files and you can close them. The second one is the `code editor` itself where you can modify your code, insert or delete breakpoints by click on the proper line in the gutter and you can see where you stopped the debugging process.
```
Comment (daeyeon) [inline README.md]: devided -> divided

File: README.md [hunk 10]
```diff
@@ -1,2 +1,170 @@
...
+
+# How to use with IoT.js
+
+If you want to use the `IoT.js Code` with the [IoT.js](https://github.com/Samsung/iotjs) do the following (this is assumes that you have a cloned IoT.js and you are in the root directory):
```
Comment (daeyeon) [inline README.md]: is assumes -> is assuming or assumes

File: README.md [hunk 11]
```diff
@@ -1,2 +1,170 @@
...
+
+```
+# Build the IoT.js with the following swtiches.
```
Comment (daeyeon) [inline README.md]: switches

File: README.md [hunk 12]
```diff
@@ -1,2 +1,170 @@
...
+$ ./build/x86_64-linux/debug/bin/iotjs --start-debug-server --jerry-debugger-port={number} {file}
+
+# To run with show opcodes.
+$ ./build/x86_64-linux/debug/bin/iotjs --start-debug-server --show-opcodes {file}
+
+# To run with source waiting mode (allows the on-the-fly source code sending).
+$ ./build/x86_64-linux/debug/bin/iotjs --start-debug-server --debugger-wait-source
+```
+
+After the `IoT.js` started running you can connect to it with the `Connect to` button or with the `connect` command in the command line and than you can start using the `IoT.js Code` debugger tools.
```
Comment (daeyeon) [inline README.md]: and then

File: README.md [hunk 13]
```diff
@@ -1,2 +1,170 @@
...
+
+# How to use with JerryScript
+
+If you want to use the `IoT.js Code` only with the [JerryScript](https://github.com/jerryscript-project/jerryscript) do the following (this is assumes that you have a cloned IoT.js and you are in the IoT.js root directory):
```
Comment (daeyeon) [inline README.md]: assumes or is assuming

File: README.md [hunk 14]
```diff
@@ -1,2 +1,170 @@
...
+$ ./build/bin/jerry --start-debug-server --debugger-wait-source
+```
+
+After the `JerryScript` started running and you have passed the `--log-level 3` runtime switch, you have to see the following message in the terminal:
+
+```
+Waiting for client connection
+```
+
+In this case you can connect to the server from the `IoT.js Code` with the `Connect to` button or with the `connect` command in the command line and you can start using the deabugger tools.
```
Comment (daeyeon) [inline README.md]: suggest a comma: In this case, you ..
typo: debugger

File: README.md [hunk 15]
```diff
@@ -1,2 +1,170 @@
...
+$ npm run eslint
+
+# To autofix the basic ESLint errors.
+$ npm run lint-autofix
+```
+
+**Note:** This project requires [node.js](https://nodejs.org/en/) v8.x.x or higher and [npm](https://www.npmjs.com) 5.x.x or higher to be able to work properly.
+
+# Layout overview
+Latest [Demo Page](https://samsung.github.io/iotjscode/).
```
Comment (daeyeon) [inline README.md]: Latest -> The latest demo or Live demo

File: README.md [hunk 16]
```diff
@@ -1,2 +1,170 @@
...
+
+There are four section in the `IoT.js Code`:
+1. **Menu**
+
+    The menu contains three more section:
+    - The hamburger style **menu toggle button**, which is for to close or open the whole menu. In closed state the menu item texts are not visible, just the icons.
+    - The **File**, **Settings** and **Download** items. In the File menu, you can open, create or save a file. In the Settings menu you can modify the editor settings or you can reset the settings to the default values. Finally in Download manu you can download the Memory Usage report if that is available.
+    - The **panel toggle buttons**, these are for to turn on or off a single panel on the information panels area.
```
Comment (daeyeon) [inline README.md]: these are for to -> `these are to`, or else, `these turn on`

File: README.md [hunk 17]
```diff
@@ -1,2 +1,170 @@
...
+In this case you can connect to the server from the `IoT.js Code` with the `Connect to` button or with the `connect` command in the command line and you can start using the deabugger tools.
+
+After you connected to the server you have to see a similar message in the terminal like this:
+```
+Connected from: 127.0.0.1
+```
+
+# License
+
+IoT.js is Open Source software under the Apache 2.0 license[Apache 2.0 license](LICENSE). Complete license and copyright information can be found within the code.
```
Comment (daeyeon) [inline README.md]: `Apache 2.0 license` is duplicated.

**Comment by knightburton** [comment]
@daeyeon Thank you! I should have been more careful.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**